great_expectations/core/expectation_suite.py

```diff
@@ -507,26 +389,7 @@ def remove_expectation(
             ]
             return result
 
-    def remove_all_expectations_of_type(
-        self, expectation_types: Union[List[str], str]
-    ) -> List[ExpectationConfiguration]:
-        if isinstance(expectation_types, str):
-            expectation_types = [expectation_types]
-
-        removed_expectations = [
-            expectation.configuration
-            for expectation in self.expectations
-            if expectation.expectation_type in expectation_types
-        ]
-        self.expectations = [
-            expectation
-            for expectation in self.expectations
-            if expectation.expectation_type not in expectation_types
-        ]
-
-        return removed_expectations
-
-    def find_expectation_indexes(
+    def _find_expectation_indexes(
```

Comment:
This is used by other methods so it has to stay around but I don't think it needs to be public

Reply:
I feel like we're misusing this. This method isn't private; it's deprecated. And it's already not in the public api. `typing-extensions` has a `deprecated` decorator. Thoughts on just using that?

Reply:
I suppose I care less about the distinction if the only external use case is in a test

Reply:
Just to close the loop on this, this method is only ever used within this class (at least after the other deletions in this file).